Automattic/wp-calypso
# BackupBrowser: Add file previews

## Proposed Changes

* Allow file previews for `text`, `code`, `audio`, `image` and `video` files. Refer to [this list](https://github.com/Automattic/wp-calypso/blob/751c0cfef75805287aafd55672f2cd2af7c730d4/client/my-sites/backup/backup-contents-page/file-browser/util.ts#L11) for the extensions mapped for each type.
* Map `*.webp` and `*.svg` files as images in the backup browser. Doing this, we will be able to display file previews for these types.
* Hide sensitive files preview. In another PR we will enable the option with a warning prompt. Discussion in p1689015874978679-slack-C059R8T8Q15.
* This PR is not including style formatting for code files.
* Previews are not available for files with 0 bytes

## Demo
https://github.com/Automattic/wp-calypso/assets/1488641/711ce71a-880f-44e6-bf33-1bce376c6485

## Screenshots

| File type | Screenshot |
|---|---|
| Image | <img width="300" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/5b1b6926-0888-4024-8277-03003db99bac"> |
| Audio | <img width="300" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/4c658d32-4eb3-4eab-95e4-45e7df35c758"> |
| Video | <img width="300" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/98a53677-22a5-47dd-839a-ef3c6b544c90"> |
| Text | <img width="300" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/ec8a0366-dfa2-48cf-8510-05d90ad2e0d4"> |
| Code | <img width="300" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/5401b6d7-2798-4d3d-b94c-8d2ae7a4d4aa"> |

## Testing Instructions
* Start a Jetpack Cloud or Calypso Live branch
* Select one of your sites with a Jetpack VaultPress Backup plan.
* Navigate to VaultPress Backup page
* Pick one date with a backup available (use the calendar and click on any date marked with a dot).
* Browse files and click on any image, video, audio, text, or code file. It should expand the file info with a `Download file` button and also the file preview.

## Pre-merge Checklist
